src/app/sold-products/sold-products.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit } from '@angular/core';
+import { Product } from '../product';
+import { ProductService } from '../product.service';
+
+@Component({
+  selector: 'app-sold-products',
+  templateUrl: './sold-products.component.html',
+  styleUrls: ['./sold-products.component.scss']
+})
+export class SoldProductsComponent implements OnInit {
+
+  products: Product[] = [];
+  targetValue: number = 0;
+
+  constructor(private productService: ProductService) { }
+
+  ngOnInit(): void {
+    this.getTargetValue();
+    this.getProducts();
+  }
+
+  getProducts(): void {
+    this.productService.getProductsData()
+      .subscribe(products => this.products = products);
```

Comment:
Same here: do not subscribe in methods. This way you loose everything you gain with Observables.
Example:
products =  this.productService.getProductsData();

In view: 
{{ products | async}}

Much cleaner + view will unsubscribe all dependencies when destroyed.

Reply:
Updated